### DISCUSSION TITLE
Why is last stable release in 2015?

### DISCUSSION THREAD

**Discussion by ab12gu** [xunit/xunit#2437] [General]
Hi, 

Why is the last stable release in 2015? 

Best, 

-ag

**Comment by bartelink** [comment]
Yes, there are no release notes posted, and no changelog.

Would you like to start contributing them?

If you actually want to simply know what the last stable release is, most people infer information like that from https://www.nuget.org/packages/xunit

Aside: there is work in progress on a V3, and a pretty clear message that no features will be added to V2.

The 'problem' is that there's a low number of contributors, and those contributors also have day jobs, so sometimes things like release notes don't happen.

**Reply by ab12gu** [reply]
I can contribute, but this leads me to believe that other test packages such as MSunit and Nunit may be better for creating C# based applications. 

They would have more support via Microsoft and re-occurring releases and updates. 

Yeah, the "problem" of low contributors is a re-occurring theme on many open-sourced projects :(.

**Reply by ab12gu** [reply]
Any ideas if its worth using MSunit or Nunit would be a better argument based on support? I know XUnit has the capabilities I like over the former two, but overall the testing development is mostly dependent on user, not framework.

**Reply by bartelink** [reply]
There are good reasons to trust in the future of all three
- NUnit has a huge number of users. Its older and has always been adding anything people ask for. I had expected it to fall into disuse 10y ago but am definitely wrong. The number of new projects picking it over xUnit is probably reducing, but some people prefer it and good for them.
- xUnit is used by tonnes of important codebases inside MS and outside. Its always been selective about adding things but has always been bullet proof and clean. The commit history tells you the bus factor is a risk, but the fact that its clean and tight will help if another maintainer would ever be needed. While some people read stuff into Brad's manner that are unfair, the bottom line is that xUnit is tight and clean and always reliable for a reason, and needs little documentation because its features are elegantly designed.
- MSunit is its own crazy box of frogs. Not many people pick it for new projects, even if it does its job perfectly fine - it ticks the same boxes. There is no reason to believe MS would ever discontinue it.

The other consideration is that Rider, VS and VSCode all support all three pretty much equally well.

I would make the same choice of using xUnit today as I did in 2009 despite that it can be argued that the number of contributors has dropped since then.

Lots of the stuff in V3 is looking good, and I'm perfectly happy for it to ship when it ships. If I had a PM over my head asking for a feature matrix or we were having some wacky team vote on testing frameworks situation, I'm sure there would be people making cases for NUnit and MSTest as being "better supported". There's a serious number of people that would be as mad as I would be in such a situation doing various projects around the world!

~I personally believe in [Keeping a Changelog](https://github.com/jet/equinox/blob/master/CHANGELOG.md), and take the time [to paste stuff from it into the GtiHub Releases](https://github.com/jet/equinox/releases) - if anyone is willing to generate one (not a big dump of commits), I'd suggest raising an Issue to see if that would be welcomed. The bottom line here is that Brad or one of the other maintainers deal with the consequences of how they choose to communicate status across GH Issues, GH Releases, Nuget and Twitter so I would not seek to impose the burden of adding release notes and/or even maintaining a Changelog file that I seed without his agreement.~ I was basing all this on hand-typing xunit.github.io as the doc site URL from memory, silly me.

**https://xunit.net/ has a full set of release notes that convey the status exactly as you anticipated; I had just forgotten about it. The TL;DR is that there have not been releases in the last 2-3 y as the focus is on developing V3 and V2 is stable and reliable.**

**Reply by ab12gu** [reply]
Hey this was super helpful, sorry about the late response. Things get busy fast. 

Yeah, the MSunit bullet was in particularly useful. It made me wonder why that was so supported with it being a kinda underwhelming compared to XUnit imo. 

also, nice cross-out. 

I'll be starting/sticking with XUnit for my current project :). Thanks!

**Comment by slaneyrw** [comment]
Is there a backlog, or feature set, that defines the minimum bar for v3.  I may be able to help progress the efforts

**Reply by Romanx** [reply]
Here's the roadmap for v3, there may be others not on this list: https://github.com/xunit/xunit/issues/2133

Some issues are also marked with help-wanted.
